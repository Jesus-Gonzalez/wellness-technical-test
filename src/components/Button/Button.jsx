import React from 'react'
import PropTypes from 'prop-types'

const propTypes = {
  submit: PropTypes.bool,
}

export const Button = props => {
  const {
    children,
    className,
    submit,
    style,
    ...attributes
  } = props

  const type = React.useMemo(() => submit ? 'submit' : 'button', [submit])
  const klass = React.useMemo(
    () => [className, 'btn', style && `btn-${style}`, !style && 'btn-primary']
      .filter(k => k).join(' '),
    [className, style]
  )

  return (
    <button
      {...attributes}
      className={klass}
      type={type}
    >
      {children}
    </button>
  )
}

Button.propTypes = propTypes
