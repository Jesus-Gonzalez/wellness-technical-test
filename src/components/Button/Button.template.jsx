import React from 'react'
import PropTypes from 'prop-types'

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.element
  ]).isRequired,
  submit: PropTypes.bool,
  className: PropTypes.string,
  type: PropTypes.string
}

export const Button = props => {
  const {
    children,
    className,
    submit,
    type,
    ...attributes
  } = props

  const buttonType = React.useMemo(() => submit ? 'submit' : 'button', [submit])
  const klass = React.useMemo(
    () => [className, 'btn', type && `btn-${type}`, !type && 'btn-primary']
      .filter(k => k).join(' '),
    [className, type]
  )

  return (
    <button
      {...attributes}
      className={klass}
      type={buttonType}
    >
      {children}
    </button>
  )
}

Button.propTypes = propTypes
