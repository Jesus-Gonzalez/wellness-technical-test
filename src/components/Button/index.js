export { Button } from './Button.template'
