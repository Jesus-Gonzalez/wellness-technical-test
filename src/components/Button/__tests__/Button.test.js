import React from 'react'
import renderer from 'react-test-renderer'

import { Button } from '../Button.template'

describe('components', () => {
  describe('Button', () => {
    it('should render', () => {
      const Component = renderer.create(<Button>buttonText</Button>)

      expect(Component).toMatchSnapshot()
    })

    it('should render className', () => {
      const Component = renderer.create((
        <Button
          className="test-class-123123"
        >
          buttonText
        </Button>
      ))

      expect(Component).toMatchSnapshot()
    })

      it('should render btn-<type>', () => {
        const Component = renderer.create((
          <Button
            type="invisible"
          >
            buttonText
          </Button>
        ))

        expect(Component).toMatchSnapshot()
      })

      it('should be submit if property added', () => {
      const Component = renderer.create((
        <Button
          submit
        >
          buttonText
        </Button>
      ))

      expect(Component).toMatchSnapshot()
    })
  })
})
