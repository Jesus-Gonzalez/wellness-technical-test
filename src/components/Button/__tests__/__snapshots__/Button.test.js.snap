// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Button should be submit if property added 1`] = `
<button
  className="btn btn-primary"
  type="submit"
>
  buttonText
</button>
`;

exports[`components Button should render 1`] = `
<button
  className="btn btn-primary"
  type="button"
>
  buttonText
</button>
`;

exports[`components Button should render btn-<type> 1`] = `
<button
  className="btn btn-invisible"
  type="button"
>
  buttonText
</button>
`;

exports[`components Button should render className 1`] = `
<button
  className="test-class-123123 btn btn-primary"
  type="button"
>
  buttonText
</button>
`;
